### ISSUE TITLE
Implications of increasing NCCL_BUFFSIZE

### ISSUE THREAD

**Issue by anirudh2290** [NVIDIA/nccl#559]
I wanted to understand the implications of changing NCCL_BUFFSIZE. We recently encountered an Out of Memory Issue for a model training with a footprint within and very close to the GPU memory limits. It was running on a container with PT1.7 and CUDA 11.0 and CUDNN 8.0.4. We switched to a different container for the model: PT1.8, CUDA 11.1.1 and CUDNN 8.0.5 and ran into OOM issue with NCCL. We were able to mitigate this issue by increasing the NCCL_BUFFSIZE to 16 as suggested in https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/env.html#nccl-buffsize. Can you please throw light on why this may have helped us with our problem ?

**Comment by szhengac** [comment]
@kwen2501 When we use `ncclGroupStart` and `ncclGroupEnd` to aggregate multiple communication operations, does each of them use different buffer?

**Comment by kwen2501** [comment]
It depends.
When you have quite a few NCCL channels and less operations to aggregate, those operations will be assigned to different channels to run concurrently. In this case, they use different intermediate buffers.
But when you have only few NCCL channels and more operations to aggregate, some operations will be serialized in the same channel. In this case, they use the same intermediate buffer, but at different time.

**Comment by szhengac** [comment]
@kwen2501 Thanks for your quick response. For the former case, do we distribute the data in a single operation into multiple free channels to increase parallelism? For the latter case, does it mean that the total buffer size is `2*NCCL_BUFFSIZE*NUM_CHANNELS` for all-reduce, reduce-scatter, and all-gather collectives? where factor `2` accounts for send and recv.

**Comment by szhengac** [comment]
@kwen2501 Have you had time to look at the above question? Thank you in advance.

**Comment by szhengac** [comment]
@kwen2501 Thanks for your detailed answer. If we have multiple communicators, I think each of them will have its own buffers?

**Comment by sjeaugey** [comment]
Yes each communicator is independent and allocates its own set of buffers.

**Comment by szhengac** [comment]
@sjeaugey thanks for your response!

**Comment by stephenmsachs** [comment]
### Issue Cleanup: Helping Us Focus on Current Challenges

  We're reviewing older issues to ensure we prioritize the most relevant and active ones. Since this issue hasn't seen updates in over 6 months, we'll be closing it for now.

  ,*This change helps us focus our efforts on addressing any current issues our users are facing.* If this issue still affects you, please don't hesitate to reopen it with a quick update (e.g., "Still relevant on [version=X]").
  Thanks for your understanding and for contributing to NCCL.
